lib/provider/song_data_preprocess.dart

```diff
@@ -0,0 +1,67 @@
+class SongDataPreprocess {
+  static RegExp _korean = new RegExp(r"^[가-힣 ]*$");
+  static RegExp _english = new RegExp(r"^[a-zA-Z ]*$");
+
+  static bool _isStartWithKorean(String title) {
+    bool ret;
+    ret = _korean.hasMatch(title.split("")[0]) ? true : false;
+
+    return ret;
+  }
+
+  /// 영어와 한국어가 섞여 있을 경우, 첫 단어를 기준으로 그에 맞는 언어로만 구성된 `String` 을 리턴
+  static String _divideLanguage(String target, bool isKorean) {
+    String korExtract = "";
+    String engExtract = "";
+
+    List<String> words = target.split("");
+
+    for (int i = 0; i < words.length; i++) {
```

Comment:
이 for loop보다 개선된 것을 사용하는게 나음
```suggestion
    for (final word in words) {
```